src/plugins/github/graphql.js

```diff
@@ -12,14 +34,64 @@ import {build} from "../../graphql/queries";
  * tune the page sizes of various entities to keep them comfortably
  * within the global capacity.
  *
+ * We use the `PAGE_LIMIT` field for the top-level page size in
+ * continuations.
+ *
  * [1]: https://developer.github.com/v4/guides/resource-limitations/#node-limit
  */
+export const PAGE_LIMIT = 100;
 const PAGE_SIZE_ISSUES = 100;
 const PAGE_SIZE_PRS = 100;
 const PAGE_SIZE_COMMENTS = 20;
 const PAGE_SIZE_REVIEWS = 10;
 const PAGE_SIZE_REVIEW_COMMENTS = 10;
 
+/**
+ * What's in a continuation? If we want to fetch more comments for the
+ * 22nd issue in the results list, we fire off the following query:
+ *
+ *    _n0: node(id: "<opaque-id-for-issue>") {
+ *      ... on Issue {
+ *        comments(first: PAGE_LIMIT, after: "<some-cursor>") {
+ *          ...comments
+ *        }
+ *    }
+ *
+ * This would be represented as:
+ *
+ *    {
+ *      enclosingNodeType: "ISSUE",
+ *      enclosingNodeId: "<opaque-id-for-issue>",
+ *      selections: [b.inlineFragment("Issue", ...)],
+ *      destinationPath: ["repository", "issues", 21],
+ *    }
+ *
+ * The `enclosingNodeId` and `selections` are used to construct the
+ * query. The `destinationPath` is used to merge the continued results
+ * back into the original results. The `enclosingNodeType` is required
+ * so that we know how to check for further continuations on the result.
+ * See function `continuationsFromContinuation` for more details on the
+ * last one.
+ *
+ * The nonce (`_n0`) is deliberately not included in the continuation
+ * type, because the nonce is a property of a particular embedding of
+ * the continuation into a query, and not of the continuation itself.
+ */
+export type Continuation = {|
+  +enclosingNodeType:
+    | "REPOSITORY"
+    | "ISSUE"
+    | "PULL_REQUEST"
+    | "PULL_REQUEST_REVIEW",
```

Comment:
It feels like there's some repetition of information between the EnclosingNodeType and other sequences, e.g. maybe the types of fragments? In principle, it would be nice if we could consolidate this all in one place / enforce the mapping at the type level. In practice, I think our attempts to do that within the GitHub plugin suggest it's not worth the effort right now. But wanted to mention it all-the-same.

Reply:
Without getting into too much detail (but happy to elaborate):
  - I agree.
  - These would be improved by a GitHub- or Relay-general API.
  - I am in general open to coalescing the various points of repetition, but think that the marginal value is very low. There are a few places where I can imagine that the types are not necessarily forced to be equal, and in those cases the cost of introducing a bad abstraction would be significant.
  - I think (agree) that doing any kind of type-level validation is not the right idea. Here, it might technically work, because everything is hard-coded—but in general, we want all the types to be dynamic, anyway, so we'd be hamstringing ourselves for much pain and no gain.